### PR TITLE
draft of FVM atomic switch FIP.

### PR DESCRIPTION
**Pull request by raulk**

### CODE REVIEW

File: FIPS/fip-atomic_fvm_switch.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+- [Incentive Considerations](#incentive-considerations)
+- [Product Considerations](#product-considerations)
+- [Implementation](#implementation)
+- [Copyright](#copyright)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Simple Summary
+
+This FIP motions for an **atomic switch** from the current intrinsic VM to a
```
Comment (Stebalien) [inline FIPS/fip-atomic_fvm_switch.md:45]: motions -> proposes?

File: FIPS/fip-atomic_fvm_switch.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+**non-programmable version of the Filecoin Virtual Machine**, at a chain epoch
+TBD.
+
+A baseline specification of the Filecoin Virtual Machine is provided in
+[filecoin-project/FIPs#288](https://github.com/filecoin-project/FIPs/pull/288).
+Only the canonical built-in actors will be supported after the atomic switch.
+
+This switch won't introduce new user-facing features or capabilities, but it's
+an important step in the trajectory towards full on-chain user-programmability.
```
Comment (Stebalien) [inline FIPS/fip-atomic_fvm_switch.md:54]: Well, I'd say it should also improve network security by ensuring that execution is deterministic and sandboxed, but I'm not sure if that counts as "user facing".

File: FIPS/fip-atomic_fvm_switch.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+## Specification
+
+### WASM bytecode management
+
+The WASM bytecode for built-in actors will be available within CAR files
+generated by the build scripts at repo:
+https://github.com/filecoin-project/canonical-actors (TBD).
+
+These CAR files will contain one block per built-in actor. The block will
+contain the bytecode in raw and full (no chunking into IPLD DAGs).
```
Comment (Stebalien) [inline FIPS/fip-atomic_fvm_switch.md:98]: IPLD block

File: FIPS/fip-atomic_fvm_switch.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+The node must load the content of these CAR files into the node's state
+blockstore. These blocks will remain orphan until the migration is run at the
+upgrade epoch. Thus, any form of stage garbage collection must explicitly
+protect these blocks from deletion (e.g. splitstore).
+
+The node must verify that exactly N blocks were loaded, with CIDs ...
+
```
Comment (Stebalien) [inline FIPS/fip-atomic_fvm_switch.md:106]: > At the migration, the actors will be linked into the state-tree.

**Review by Stebalien — APPROVED** [review]
Nothing controversial here, really. But I think we'll need to modify the init actor a bit. It needs to know what actor types exist.

**Comment by raulk** [comment]
Submitted publicly: https://github.com/filecoin-project/FIPs/pull/294.

**Pull request closed without merging**